core/rpc/client/user/http/convert.go

```diff
@@ -216,40 +226,229 @@ func convertHttpForeignKeys(foreignKeys []httpTx.TxForeignKey) []*transactions.F
 	for i, action := range actions {
-		acts[i] = &transactions.Action{
+		mods := make([]types.Modifier, len(action.Modifiers))
+		for j, mod := range action.Modifiers {
+			mods[j] = types.Modifier(mod)
+		}
+
+		acts[i] = &types.Action{
 			Name:        action.Name,
 			Annotations: action.Annotations,
-			Inputs:      action.Inputs,
-			Mutability:  action.Mutability,
-			Auxiliaries: action.Auxiliaries,
+			Parameters:  action.Parameters,
 			Public:      action.Public,
-			Statements:  action.Statements,
+			Modifiers:   mods,
+			Body:        action.Body,
 		}
 	}
 	return acts
 }
 
-// convertHttpExtensions converts []httpTx.TxExtension to []transactions.Extension
-func convertHttpExtensions(extensions []httpTx.TxExtensions) []*transactions.Extension {
-	exts := make([]*transactions.Extension, len(extensions))
+func convertHttpExtensions(extensions []httpTx.TxExtensions) []*types.Extension {
+	exts := make([]*types.Extension, len(extensions))
 	for i, extension := range extensions {
-		initialize := make([]*transactions.ExtensionConfig, len(extension.Initialization))
+		initialize := make([]*types.ExtensionConfig, len(extension.Initialization))
 		for j, init := range extension.Initialization {
-			initialize[j] = &transactions.ExtensionConfig{
-				Argument: init.Argument,
-				Value:    init.Value,
+			initialize[j] = &types.ExtensionConfig{
+				Key:   init.Argument,
+				Value: init.Value,
 			}
 		}
 
-		exts[i] = &transactions.Extension{
-			Name:   extension.Name,
-			Config: initialize,
-			Alias:  extension.Alias,
+		exts[i] = &types.Extension{
+			Name:           extension.Name,
+			Initialization: initialize,
+			Alias:          extension.Alias,
 		}
 	}
 	return exts
 }
+
+func convertHttpProcedures(procedures []httpTx.TxProcedure) []*types.Procedure {
+	procs := make([]*types.Procedure, len(procedures))
+	for i, procedure := range procedures {
+		mods := make([]types.Modifier, len(procedure.Modifiers))
+		for j, mod := range procedure.Modifiers {
+			mods[j] = types.Modifier(mod)
+		}
+
+		var returns *types.ProcedureReturn
+		if len(procedure.ReturnTypes) > 0 {
+			returns = &types.ProcedureReturn{}
+
+			tps := make([]*types.DataType, len(procedure.ReturnTypes))
+			for j, tp := range procedure.ReturnTypes {
+				tps[j] = convertDataType(&tp)
+			}
+
+			returns.Types = tps
+		} else if procedure.ReturnTable != nil && len(procedure.ReturnTable) > 0 {
+			returns = &types.ProcedureReturn{}
+
+			cols := make([]*types.NamedType, len(procedure.ReturnTable))
+			for j, col := range procedure.ReturnTable {
+				cols[j] = &types.NamedType{
+					Name: col.Name,
+					Type: convertDataType(col.Type_),
+				}
+			}
+
+			returns.Table = cols
+		}
+
+		procs[i] = &types.Procedure{
+			Name:        procedure.Name,
+			Annotations: procedure.Annotations,
+			Modifiers:   mods,
+			Public:      procedure.Public,
+			Parameters:  convertHttpParameters(procedure.Parameters),
+			Body:        procedure.Body,
+		}
```

Comment:
Need to `Returns: returns`?  The `returns` prepared above seems to be discarded.